vars/neoDeploy.groovy

```diff
@@ -216,7 +231,73 @@ void call(parameters = [:]) {
                 lock("$STEP_NAME:${neoCommandHelper.resourceLock()}") {
                     deploy(script, configuration, neoCommandHelper, configuration.dockerImage, deployMode)
                 }
+                inValidateCache(configuration)
+            }
+        }
+    }
+}
+
+private inValidateCache(configuration){
+    if(configuration.neo.invalidateCache == true) {
+        def account = configuration.neo.account
+        def host = configuration.neo.host
+
+        withCredentials([usernamePassword(
+            credentialsId: configuration.neo.oauthCredentialId,
+            passwordVariable: 'OAUTH_NEO_CLIENT_SECRET',
+            usernameVariable: 'OAUTH_NEO_CLIENT_ID')]) {
+            def bearerTokenResponse = sh(
+                script: """#!/bin/bash
+                   curl -X POST -u "${OAUTH_NEO_CLIENT_ID}:${OAUTH_NEO_CLIENT_SECRET}" \
+           \"https://oauthasservices-${account}.${host}/oauth2/api/v1/token?grant_type=client_credentials&scope=write,read\"
+                """,
+                returnStdout: true
+            )
+            def fetchBearerTokenJson = readJSON  text: bearerTokenResponse
+
+            def bearerToken = fetchBearerTokenJson.access_token
+
+            echo "Retrieved bearer token."
+
+            def xcsrfTokenResponse = sh(
+                script: """#!/bin/bash
+                               curl -i -L \
+                               -c 'cookies.jar' \
+                               -H 'X-CSRF-Token: Fetch' \
+                               -H "Authorization: Bearer ${bearerToken}" \
+                            \"https://sandboxportal-${account}.${host}/fiori/api/v1/csrf\"
+                    """, returnStdout: true)
+
+            def xcsrfTokenHeaderMatcher=xcsrfTokenResponse =~ /(?m)^X-CSRF-Token: ([0-9A-Z]*)$/
+            def xcsrfToken = xcsrfTokenHeaderMatcher[0][1]
+
+            def siteId = configuration.neo.siteId ? "${configuration.neo.siteId}" : ""
+            if(siteId ==""){
+                echo "Using the default value of siteId from Portal service and invalidating the cache."
             }
+            else{
+                echo "Invalidating the cache for site Id : ${siteId}."
+            }
+                def statusCode = sh(
+                    script: """#!/bin/bash
+                       curl -X POST -L \
+                       -b 'cookies.jar'  \
+                       -H "X-CSRF-Token: ${xcsrfToken}" \
+                       -H "Authorization: Bearer ${bearerToken}" \
+                       -d "{\"siteId\":${siteId}}" \
+                       -so /dev/null \
+                       -w '%{response_code}' \
+                       \"https://sandboxportal-${account}.${host}/fiori/v1/operations/invalidateCache\"
```

Comment:
Can we hard code `sandboxportal-` ?

Reply:
As far as I understood, it is how the api call looks. In that case, we can hardcode it.